nix/pkgs/haskell/haskell.nix

```diff
@@ -53,10 +54,18 @@ let
       {
         reinstallableLibGhc = false;
         packages = {
-          # See https://github.com/input-output-hk/plutus/issues/1213
-          marlowe.doHaddock = false;
-          plutus-use-cases.doHaddock = false;
-          plutus-ledger.doHaddock = false;
+          # See https://github.com/input-output-hk/plutus/issues/1213 and
+          # https://github.com/input-output-hk/plutus/pull/2865.
+          marlowe.doHaddock = deferPluginErrors;
```

Comment:
i'm not super happy with the use of `deferPluginErrors` here. I suppose these settings are only relevant for the creation of the documentation right? If that is the case it would be preferable to not have these changes here at all but to override them where this change is relevant.

So I guess this should be happening in plutus-haddock-combined/default.nix instead:

```nix
let
  toHaddock = haskell-nix.haskellLib.collectComponents' "library" haskell.projectPackages;
in
```
here we can modify `haskell.projectPackages` to make the changes instead:

```
nix-repl> (pkgs.haskell-nix.haskellLib.collectComponents' "library" plutus.haskell.projectPackages).marlowe.config.doHaddock
false
```

i don't have that much experience with haskell.nix but if anything i say doesn't make sense i'm sure @michaelpj will point that out ;)

Reply:
@gilligan yes absolutely we only need this for the documentation; okay i think i like your idea, so i'll give it a go; but it does have the downside that we need to explicitly list the projects twice? (once in haskell.nix and again here in the docs area)

Reply:
> but it does have the downside that we need to explicitly list the projects twice?

That is something we should avoid. Instead we should be overriding (see https://nixos.org/manual/nixpkgs/stable/#sec-pkg-overrideAttrs) the relevant derivations.

@michaelpj do you happen to know if there is any convenience functionality to apply overrides to `haskell.projectPackages` ?

Reply:
I don't think there's an overriding functionality. I think it's fine to do:
```
project = import ./haskell { ... deferPluginErrors = false; ... }
projectAllHaddock = import ./haskell { ... deferPluginErrors = true; ...}
```
This won't result in any more rebuilding than necessary, since the haddock is built in a separate derivation. We *will* rebuild the packages that have different flags (and things that depend on them), but that's not *so* bad, and I don't think there's a way around it so long as we're controlling this behaviour with cabal flags.